rpcs3/rpcs3qt/memory_string_searcher.cpp

```diff
@@ -35,48 +72,255 @@ memory_string_searcher::memory_string_searcher(QWidget* parent)
 
 void memory_string_searcher::OnSearch()
 {
-	const std::string wstr = m_addr_line->text().toStdString();
-	const char *str = wstr.c_str();
-	const u32 len = ::size32(wstr);
+	std::string wstr = m_addr_line->text().toStdString();
 
-	gui_log.notice("Searching for string %s", str);
+	if (wstr.empty() || wstr.size() >= 4096u)
+	{
+		gui_log.error("String is empty or too long (size=%u)", wstr.size());
+		return;
+	}
 
-	// Search the address space for the string
-	u32 strIndex = 0;
-	u32 numFound = 0;
-	const auto area = vm::get(vm::main);
-	for (u32 addr = area->addr; addr < area->addr + area->size; addr++)
+	gui_log.notice("Searching for %s", wstr);
+
+	const int mode = std::max(m_cbox_input_mode->currentIndex(), 0);
+	bool case_insensitive = false;
+
+	// First characters for case insensitive search
+	const char first_chacarters[2]{ static_cast<char>(::tolower(wstr[0])), static_cast<char>(::toupper(wstr[0])) };
+	std::string_view insensitive_search{first_chacarters, 2};
```

Comment:
```suggestion
	const char first_characters[2]{ static_cast<char>(::tolower(wstr[0])), static_cast<char>(::toupper(wstr[0])) };
	std::string_view insensitive_search{first_characters, 2};
```
and any following usage

Reply:
Fixed.